--- a/Cargo.toml
+++ b/Cargo.toml
@@ -61,8 +61,8 @@
     "filesystem-assets",
 ]
 
-[dependencies.proc-macro-error2]
-version = "2.0.1"
+[dependencies.proc-macro-error3]
+version = "3.1.1"
 
 [dependencies.proc-macro2]
 version = "1.0"
--- a/src/lib.rs
+++ b/src/lib.rs
@@ -3,7 +3,7 @@
 use fluent_syntax::ast::{CallArguments, Expression, InlineExpression, Pattern, PatternElement};
 use i18n_embed::{fluent::FluentLanguageLoader, FileSystemAssets, LanguageLoader};
 use proc_macro::TokenStream;
-use proc_macro_error2::{abort, emit_error, proc_macro_error};
+use proc_macro_error3::{abort, emit_error, proc_macro_error};
 use quote::quote;
 use std::{
     collections::{HashMap, HashSet},
