diff --git a/src/apprt/action.zig b/src/apprt/action.zig
index f6865af..0df18dc 100644
--- a/src/apprt/action.zig
+++ b/src/apprt/action.zig
@@ -455,7 +455,8 @@ pub const Action = union(Key) {
         // At the time of writing, we don't promise ABI compatibility
         // so we can change this but I want to be aware of it.
         assert(@sizeOf(CValue) == switch (@sizeOf(usize)) {
-            4 => 16,
+            // Scrollbar contains three uint64_t fields on 32-bit hosts too.
+            4 => 24,
             8 => 24,
             else => unreachable,
         });
